### DISCUSSION TITLE
Add Compatibility of Tesla Gen3 Wall Connector

### DISCUSSION THREAD

**Discussion by irigo2288** [teslamate-org/teslamate#2504] [Ideas]
Hello,

A lot of Tesla user is also using a tesla connector, cheap and functionnal with most of electrical infrastructure in EU and US.
The new generation Gen3 give api information. 

It can be interesting to develop a module for monitoring the consomation KWH of each session.
The connection with the Tesla wall connector Gen3 is easy, it's a non-authentification dialog to the IP.
A simple API call give data: 
http://IP-CONNECTOR/api/1/vitals
{
    "contactor_closed": false,
    "vehicle_connected": false,
    "session_s": 0,
    "grid_v": 237.9,
    "grid_hz": 49.864,
    "vehicle_current_a": 0.4,
    "currentA_a": 0.3,
    "currentB_a": 0.4,
    "currentC_a": 0.3,
    "currentN_a": 0.8,
    "voltageA_v": 2.0,
    "voltageB_v": 0.0,
    "voltageC_v": 2.7,
    "relay_coil_v": 11.9,
    "pcba_temp_c": 13.4,
    "handle_temp_c": 10.1,
    "mcu_temp_c": 20.0,
    "uptime_s": 21074,
    "input_thermopile_uv": -222,
    "prox_v": 0.0,
    "pilot_high_v": 11.9,
    "pilot_low_v": 11.9,
    "session_energy_wh": 0.000,
    "config_status": 5,
    "evse_state": 1,
    "current_alerts": []
}
http://IP-CONNECTOR/api/1/lifetime
{
    "contactor_cycles": 34,
    "contactor_cycles_loaded": 0,
    "alert_count": 268,
    "thermal_foldbacks": 0,
    "avg_startup_temp": 30.0,
    "charge_starts": 34,
    "energy_wh": 81242,
    "connector_cycles": 7,
    "uptime_s": 267123,
    "charging_time_s": 38835
}

i think it can give to tesla user a real view of it's consomation.

**Comment by faekz0r** [comment]
Great idea and seems like it would not be too hard to implement. 🤓

**Comment by zikronix** [comment]
theres a similar project https://github.com/rnhurt/tesla-wc/commit/067b15168b0197508d6ec1df7d840e8b523b0b44
